static/app/components/events/featureFlags/eventFeatureFlagList.tsx

```diff
@@ -68,24 +71,54 @@ export function EventFeatureFlagList({
   const {closeDrawer, isDrawerOpen, openDrawer} = useDrawer();
   const viewAllButtonRef = useRef<HTMLButtonElement>(null);
   const organization = useOrganization();
+  const eventView = useIssueDetailsEventView({group});
+  const {data: rawFlagData} = useOrganizationFlagLog({
+    organization,
+    query: {
+      start: eventView.start,
+      end: eventView.end,
+      statsPeriod: eventView.statsPeriod,
+    },
+  });
 
-  // Transform the flags array into something readable by the key-value component
-  const hydrateFlags = (flags: FeatureFlag[] | undefined): KeyValueDataContentProps[] => {
-    if (!flags) {
-      return [];
-    }
-    return flags.map(f => {
-      return {
-        item: {key: f.flag, subject: f.flag, value: f.result.toString()},
-      };
-    });
-  };
+  const {
+    suspectFlags,
+    isError: isSuspectError,
+    isPending: isSuspectPending,
+  } = useSuspectFlags({
+    organization,
+    firstSeen: group.firstSeen,
+    rawFlagData,
+    event,
+  });
 
-  // Reverse the flags to show newest at the top by default
-  const hydratedFlags = useMemo(
-    () => hydrateFlags(event.contexts?.flags?.values.reverse()),
-    [event]
-  );
+  const suspectFlagNames = useMemo(() => {
+    return isSuspectError || isSuspectPending ? [] : suspectFlags.map(f => f.flag);
+  }, [isSuspectError, isSuspectPending, suspectFlags]);
+
+  const hydratedFlags = useMemo(() => {
+    // Transform the flags array into something readable by the key-value component
+    const hydrateFlags = (
+      flags: FeatureFlag[] | undefined
+    ): KeyValueDataContentProps[] => {
```

Comment:
i don't think it makes sense to make a function inside your use memo that is then immediately called

Reply:
might want to be calling [`toReversed`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toReversed) or splat the values out first... but instead of the function something like this:

```suggestion
  const flags: FeatureFlag[] = [...event.contexts?.flags?.values].reverse();
```
the splat should remove the need for undefined in the type ^ (might not need the explicit type on the const either)

and then the next statement would be `if (!flags) { return []; } ...`